src/bin/server.ts

```diff
@@ -46,19 +47,44 @@ interface Args {
   if (configPath) {
     try {
       options = require(configPath);
-      // make sure that networkKey is passed as buffer.
-      // accept both zwave2mqtt format as ozw format
-      if (options.networkKey && options.networkKey.length === 32) {
-        options.networkKey = Buffer.from(options.networkKey, "hex");
-      } else if (options.networkKey && options.networkKey.includes("0x")) {
-        options.networkKey = options.networkKey
-          .replace(/0x/g, "")
-          .replace(/, /g, "");
-        options.networkKey = Buffer.from(options.networkKey, "hex");
-      } else {
-        console.error("Error: Invalid networkKey defined");
-        return;
+      // If both securityKeys.S0_Legacy and networkKey are defined, throw an error.
+      if (options.securityKeys?.S0_Legacy && options.networkKey) {
+        throw new Error(
+          "Both `networkKey` and `securityKeys.S0_Legacy` options are both present in the " +
+            "config and we only need one. `securityKeys.S0_Legacy` is preferred over `networkKey` " +
+            "and should match your old `networkKey` if you were using it before."
+        );
       }
+      const securityKeyNames = [
+        "S0_Legacy",
+        "S2_AccessControl",
+        "S2_Authenticated",
+        "S2_Unauthenticated",
+      ];
+      // We prefer the securityKeys option over the networkKey one
+      if (options.securityKeys)
+        for (const key of securityKeyNames)
+          if (key in options.securityKeys)
+            options.securityKeys[key] = normalizeKey(
+              options.securityKeys[key],
+              `securityKeys.${key}`
+            );
+      // If we get here, securityKeys.S0_Legacy is not defined, so we can safely use networkKey
+      // make sure that networkKey is passed as buffer and accept both zwave2mqtt format and ozw format
+      if (options.networkKey) {
+        options.securityKeys.S0_Legacy = normalizeKey(
+          options.networkKey,
+          "networkKey",
+          true
+        );
+        console.warn(
+          "The `networkKey` option is deprecated in favor of `securityKeys` option. To eliminate " +
```

Comment:
Does it make sense to create an object for this and not just have option `networkKey`, `networkKeyS2AccessControl` etc ? That way we don't have to deprecate the S0 one and can just keep it as `networkKey`?

Reply:
I thought the intention here was to follow the `zwave-js` [ZWaveOptions schema](https://zwave-js.github.io/node-zwave-js/#/api/driver?id=zwaveoptions). The only reason we have to transform the keys is because the driver expects a buffer which can't be expressed in JSON.

I personally think we should stick with this approach, but maybe that's just me

Reply:
Good point! We should follow that schema.